src/NaCl.Core/Base/Snuffle.cs

```diff
@@ -22,13 +22,13 @@ public abstract class Snuffle
     {
         protected const int KEY_SIZE_IN_INTS = 8;
         public const int KEY_SIZE_IN_BYTES = KEY_SIZE_IN_INTS * 4; // 32
-        protected const int BLOCK_SIZE_IN_INTS = 16;
+        protected internal const int BLOCK_SIZE_IN_INTS = 16;
         public const int BLOCK_SIZE_IN_BYTES = BLOCK_SIZE_IN_INTS * 4; // 64
 
         protected static uint[] SIGMA = new uint[] { 0x61707865, 0x3320646E, 0x79622D32, 0x6B206574 }; // "expand 32-byte k" (4 words constant: "expa", "nd 3", "2-by", and "te k")
 
         protected readonly ReadOnlyMemory<byte> Key;
-        protected readonly int InitialCounter;
+        internal protected readonly int InitialCounter;

```

Comment:
Something minor... the access modifier should be `protected internal`, just need to fix the ordering.